### PR TITLE
Added examples and fixed documentation bugs

### PR DESCRIPTION
**Pull request by madchutney**

### CODE REVIEW

File: docs/news/1374.feature [hunk 1]
```diff
@@ -1 +1 @@
-Support for device group
\ No newline at end of file
+Support for added for Device Groups.
```
Comment (acabarbaye) [inline docs/news/1374.feature]:
```suggestion
Support added for Device Groups.
```

File: tests/integration/examples/test_resource_values.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+@BaseCase._skip_in_ci
+class TestExamples(BaseCase):
+
+    def test_get_and_set_resource_value(self):
+        # an example: get and set a resource value
+        from mbed_cloud.foundation import Device
+        from mbed_cloud import ConnectAPI
+
+        # Use the Foundation interface to find a device.
+        device = Device().list(max_results=1).next()
```
Comment (acabarbaye) [inline tests/integration/examples/test_resource_values.py]: it needs to be a connected device, so we need to set a filter device_state__eq=REGISTERED

File: tests/integration/examples/test_resource_values.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+        # Use the Legacy interface for find resources
```
Comment (acabarbaye) [inline tests/integration/examples/test_resource_values.py]:
```suggestion
        # Use the Legacy interface to find resources
```

File: tests/integration/examples/test_resource_values.py (hunk 2 above)
Comment (madchutney) [reply tests/integration/examples/test_resource_values.py]: Done

**Pull request merged**